### ISSUE TITLE
[Feature] Obtain only variable description

### ISSUE THREAD

**Issue by Letme** [michaeljones/breathe#216]
For example the only-description flag in struct member should just print the description from that member. Only-name would be useful to get only name of that variable. This is so that we can customize the output in rst for example in table with only requested properties.

example.c
```c
struct example {
    int member1; /**< First member of example struct */
    int member2; /**< Second member of example struct */
    int member3; /**< Third member of example struct */
};
```

example.rst
```
The example struct member1 (
.. doxygenstruct:: <struct name>
   :member: <member name>
   :only-description:
) is responsible for making the example go mad
```

which should render to 
```
The example struct member1 (First member of example struct) is responsible for making example go mad
```

**Comment by vitaut** [comment]
I don't think you can use directives as inline markup in Sphinx. This should probably be a Sphinx role instead, something like

```rst
The example struct member1 (:member-description:`example::member1`)
is responsible for making the example go mad
```

**Comment by Letme** [comment]
My idea is a bit more wide than just that simple example, but I would just like to get description from breathe of certain struct member to add it to .rst table which I am making with other script.

**Comment by michaeljones** [comment]
I would agree with @vitaut that it should be a role if it has to be inline in text. If it is the only thing in the table cell then you might get away with a directive. Though automatically formatting a table with a directive in it won't be that much fun! :) It would be good to fully understand what you need and if this is something that will help other people as well. Could you provide a little more information?

**Comment by Letme** [comment]
So inline directive might form a problem, but full blown table not? I would be doing below rst tables instead of normal presentations without size types/sizes, etc. Before or after the variable (or struct member in this case) name I would append my own columns if/when needed to provide additional information. That is why I thought inlining would be useful - as with that we can form whatever we wish from doxygen in whatever design we fit. I have a script to implement a table in rst I would contribute, but I would then just need some advice how to get variable names and descriptions from breathe in rst.

```rst
+----------+-----------------+---------------------------------+
| size     | variable name   | variable description            |
+==========+=================+=================================+
| 32 bytes | example:member1 | First member of example struct  |
+----------+-----------------------------+---------------------+
| 32 bytes | example:member2 | Second member of example struct |
+----------+-----------------------------+---------------------+
| 32 bytes | example:member3 | Third member of example struct  |
+----------+-----------------------------+---------------------+
```

**Comment by rweickelt** [comment]
Do I get You right: What You basically want is some kind of template
mechanism to customize the output of directives? For instance to create Your
own class, namespace, group, and whatever index pages instead of following a
fixed structure?

Some thoughts about Your above example: There is documentation for people
how "use" the code and there is documentation for people who are supposed to
work on it; a big difference in my opinion. Users would likely not digg into
the code, but prefer a clean and neat html documentation. In that case, it
is not just some kind of by-product, but an indispensable part needing the
same level of attention and effort as the source code itself.

It is one way to put sophisticated documentation as near as possible to the
source code, e.g. description beside struct members (does literate
programming ring a bell?), but it often makes the code harder to read, too
and bloats header files.
So, if the table from above is meant for "users", then just see it as an
integral document and write it separately in rst or doxygen table markup. If
it is ment for developers who make changes to the code, don't waste Your
time with an elaborate html documentation. They are experts and required to
know the user documentation anyway. Keep the code readable and sprinkle
(carefully) some non-doxygen comments into it where it is not self-explaining.

**Comment by Letme** [comment]
You get it right, but I would not agree with your last two paragraphs. I write doxygen comments really strictly, commenting each struct member what it should contain/be used for. Why would I then write that same documentation in .rst again just because I would like a bit more customized output? That is the whole point of breathe - to get doxygen comments inside the rst documentation without having to explain again what certain functions/structs/variables are used for. 

I would simply like to expand the formatting option to tables with a bit more customization to actually displayed values and parameters. Since inline-ing is not an option then argument would be that current representation takes a lot of space, which could be more neatly presented via tables. Also sometimes I would like to see the size of certain struct and tables make a nice representation of that and tables present an easy overview.

**Comment by duchengyao** [comment]
I also think it's a great idea to enhance the output for members of function.

Like `Public Types` and `Public Member Functions` in [https://docs.opencv.org/4.5.2/d1/dfb/classcv_1_1ml_1_1EM.html](https://docs.opencv.org/4.5.2/d1/dfb/classcv_1_1ml_1_1EM.html)
